GitSearch/Application/SearchResult/SearchResultCell.swift

```diff
@@ -112,7 +34,93 @@ class SearchResultCell: UICollectionViewCell, UICollectionViewDelegateFlowLayout
         return layoutAttributes
     }
     
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupUI()
+    }
+    
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 }
+
+extension SearchResultCell {
+
+    func fillInfo(for item: SearchItem, isLast: Bool) {
+        self.lastCell = isLast
+        self.title.text = item.name
+        self.author.text = "by \(item.owner?.login ?? "")"
+        self.subTitle.text = """
+        \(item.description ?? "")
+        """
+    }
+}
+
+private extension SearchResultCell {
+    
+    func setupUI() {
+        let arrangedSubviews = setupSubviews()
+        arrangedSubviews.forEach{contentView.addSubview($0)}
+        contentView.setNeedsUpdateConstraints()
+    }
+    
+    func setupSubviews() -> [UIView] {
+        title = {
+            let label = UILabel()
+            label.translatesAutoresizingMaskIntoConstraints = false
+            label.font = UIFont.init(name: "Avenir-Black", size: 20)
+            label.textColor = UIColor.black
+            return label
+        }()
```

Comment:
It does not make sense to use blocks here (it made sense before when you declared instance variables and immediately initialised them). But here nothing precludes you from regular code:
```suggestion
        title = UILabel()
        title.translatesAutoresizingMaskIntoConstraints = false
        title.font = UIFont.init(name: "Avenir-Black", size: 20)
        title.textColor = UIColor.black
```

Reply:
I used regular code usually, but after a while I found out that such blocks looks better for me

- property declaration are grouped 
- easy to read and fill up
- easy to distinguish from other parts

Is it bad, can you give me some link to read about relevant using of such initializationd and regular code?

Reply:
I understand your reasoning, but its non-standard approach. If you want to group related declaration, it would be much better to do something like the following:
```
class Foo {
    var mySuperLabel = {
        let label = UILabel()
        // setup code here
        return label
    }()
}
```

In this way, your declaration are not only grouped, but also separated, and you do not have a monstrous functions which setups huge list of UI elements.

Reply:
On the other hand, its somewhat a matter of taste, so if you prefer to leave it that way, I won't insist to change that.

Reply:
I choose just to separate into different files components initialization and constrains setup.
refactored components init as you propose